### ISSUE TITLE
0.68 regression: Word boundaries are not respected during search for the word under cursor

### ISSUE THREAD

**Issue by yegor-alexeyev** [vrapper/vrapper#747]
Summary: Searching for the current word(* and # shortcuts in normal mode) functionality is broken.
Symptom: Word boundaries are not respected during search
Reproduction:
Have text below in buffer:  "spread spreadsheet spread"
Enter the normal mode and place the cursor on the first word.
Press asterisk('*').
Observed behaviour: cursor is moved to the second word.
Expected behaviour: cursor is moved to the third word.

This is a software regression after updgrading to 0.68.

**Comment by keforbes** [comment]
Yep, I can reproduce this.  I'm not sure how it broke but I'll get a fix to you as soon as I can.

**Comment by keforbes** [comment]
I broke this with my fix for #739.  The Java regex engine allows for a `wholeWord` flag which I had been sending to it for this feature but my fix for #739 changed it so I'm using my own custom regex to define word boundaries based on the `iskeyword` property.  Unfortunately, I tweaked the underlying regex after performing the first search ( :disappointed: ).  So after hitting `*`, if you go into search mode `/` and hit the up arrow (back in history) you'll see your search with the correct regex for word boundaries `\< \>`.  That search will work correctly, but *only if you have regexsearch enabled*.  I need a solution where my custom regex is always run even if `regexsearch` is disabled and also runs that first time.  Sorry, just rambling here.
